pkg/netpol/eval/exposure.go

```diff
@@ -68,12 +68,19 @@ func (pe *PolicyEngine) extractLabelsAndRefineRepresentativePeers(podObj *k8s.Po
 
 // refineRepresentativePeersMatchingLabels removes from the policy engine all representative peers
 // with labels matching the given labels of a real pod
+// representative peers matching any-namespace will not be removed.
 func (pe *PolicyEngine) refineRepresentativePeersMatchingLabels(realPodLabels, realNsLabels map[string]string) {
 	keysToDelete := make([]string, 0)
 	// look for representative peers with labels matching the given real pod's (and its namespace) labels
 	for key, peer := range pe.representativePeersMap {
 		potentialPodSelector := labels.SelectorFromSet(labels.Set(peer.Pod.Labels))
 		potentialNsSelector := labels.SelectorFromSet(labels.Set(peer.PotentialNamespaceLabels))
+		if potentialNsSelector.Empty() { // representative peer that matches any-namespace, will not be removed
```

Comment:
```suggestion
		if potentialNsSelector.Empty() { 
		// empty --representative peer that matches any-namespace, thus will not be removed
		// note that if the policy had nil namespaceSelector, it would be converted to the namespace of the policy 
```